src/plugins/crypto/gpg.c

```diff
@@ -784,25 +784,37 @@ int CRYPTO_PLUGIN_FUNCTION (gpgCall) (KeySet * conf, Key * errorKey, Key * msgKe
 		if (msgKey)
 		{
 			close (STDIN_FILENO);
-			dup (pipe_stdin[0]);
+			if (dup (pipe_stdin[0]) < 0)
+			{
+				ELEKTRA_SET_ERROR (ELEKTRA_ERROR_CRYPTO_GPG, errorKey, "failed to redirect stdin.");
+				exit (42);
```

Comment:
Is 42 not used by gpg?

Reply:
Sadly GPG does not document it's return values/error codes very well. See [man-page](https://www.gnupg.org/documentation/manpage.html)